src/freenas/usr/local/sbin/ix-update

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash -e
+
+readonly UPDATE_FAILED_LOG="/data/update.failed"
+
+if [ ! -f /data/need-update ]
+then
+    exit 0
+fi
+
+has_alembic=$(echo "SELECT name FROM sqlite_master WHERE name = 'alembic_version'" | sqlite3 /data/freenas-v1.db)
```

Comment:
Are we supporting upgrades from 11.3 -> TN Scale ? If no, I think it's safe to assume that we are going to have alembic ?

Reply:
Yes, we are supporting these, that's why we have freenas-migrate113 debian package :)

Reply:
@william-gr why?

Reply:
Big jumps like that are the recipe for failure. We simply cant properly support that.
We need to eventually add some coded to 12 for hardware dependent bits to allow a smooth upgrade (e.g. interface card names) that cannot be made only through db migration.
That also includes HA/CARP

Reply:
We already have that types of migrations, e.g. https://github.com/freenas/freenas/pull/4524

Reply:
No, thats not what I meant.

For example, interfaces are identified by name. They are not the same in SCALE. We need someway to identify when being OS agnostic, e.g. mac address.
We dont have that in 11.2 or 11.3. And we cannot do that only via db migration, we first need to run middlewared on 12 to do that kind of thing before loading SCALE.
We cannot load a 11.2 or 11.3 database into SCALE and expect it to work.